core/src/main/java/org/infinispan/persistence/sifs/NonBlockingSoftIndexFileStore.java

```diff
@@ -322,27 +364,31 @@ private void migrateFromOldFormat(FileProvider oldFileProvider) {
 
    private void buildIndex(final AtomicLong maxSeqId) {
       Flowable<Integer> filePublisher = filePublisher();
-      CompletionStage<Void> stage = handleFilePublisher(filePublisher.doAfterNext(file -> compactor.completeFile(file, -1)), false, false,
-            (file, offset, size, serializedKey, entryMetadata, serializedValue, serializedInternalMetadata, seqId, expiration) -> {
-               long prevSeqId;
-               while (seqId > (prevSeqId = maxSeqId.get()) && !maxSeqId.compareAndSet(prevSeqId, seqId)) {
-               }
-               Object key = marshaller.objectFromByteBuffer(serializedKey);
-               if (log.isTraceEnabled()) {
-                  log.tracef("Loaded %d:%d (seqId %d, expiration %d)", file, offset, seqId, expiration);
-               }
-               int segment = keyPartitioner.getSegment(key);
-               // We may check the seqId safely as we are the only thread writing to index
-               if (isSeqIdOld(seqId, segment, key, serializedKey)) {
-                  index.handleRequest(IndexRequest.foundOld(segment, key, serializedKey, file, offset));
+      CompletionStage<Void> stage = filePublisher.flatMap(outerFile -> {
+         ByRef.Long nextExpirationTime = new ByRef.Long(-1);
+         return handleFilePublisher(filePublisher, false, false,
+               (file, offset, size, serializedKey, entryMetadata, serializedValue, serializedInternalMetadata, seqId, expiration) -> {
+                  long prevSeqId;
+                  while (seqId > (prevSeqId = maxSeqId.get()) && !maxSeqId.compareAndSet(prevSeqId, seqId)) {
+                  }
+                  Object key = marshaller.objectFromByteBuffer(serializedKey);
+                  if (log.isTraceEnabled()) {
+                     log.tracef("Loaded %d:%d (seqId %d, expiration %d)", file, offset, seqId, expiration);
+                  }
+                  // Make sure to keep track of the lowest expiration that isn't -1
+                  nextExpirationTime.set(nextExpirationTime.get() == -1 ? expiration : Math.min(nextExpirationTime.get(), expiration));
```

Comment:
`nextExpirationTime.set(ExpiryHelper.mostRecentExpirationTime(nextExpirationTime.get(), expiration));`?

Reply:
Yup, added this method after and forgot about this reference.